### DISCUSSION TITLE
How can I exclude the generated code from tsc when I am typechecking?

### DISCUSSION THREAD

**Discussion by rqbazan** [dotansimha/graphql-code-generator#8776] [Q&A]
Hi there,

Recently, I noticed that my CI is taking more time than usual. After some debugging I realized the problem:

```json
"check:types": "tsc --noEmit"
```

This npm script is taking too long to finish because the generated code is massive. So, how can I exclude it from `tsc` properly? 🤔  Probably it's a common problem but I don't see some issue or discussion about it.

**Comment by saihaj** [comment]
I mean you usually would want to typecheck the usage to get benefits of static type checking. As codebase grows `tsc` does slow down. There are improvement that can be done to reduce the amount of types generated.


But if you want to exclude you can just use [tsconfig exclude option](https://www.typescriptlang.org/tsconfig#type-exclude)
